src/main/webapp/components/Security.js

```diff
@@ -0,0 +1,278 @@
+
+export const Security = hh(class Security extends Component {
+
+  state = {};
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      formData: {
+        pii: '',
+        compliance: '',
+        sensitive: '',
+        accessible: '',
+        textCompliance: '',
+        textSensitive: '',
+        textAccessible: '',
+      },
+      errors: {
+        pii: true,
+        compliance: true,
+        sensitive: true,
+        accessible: true,
+        textCompliance: true,
+        textSensitive: true,
+        textAccessible: true,
+      }
+    };
+  }
+
+  componentDidCatch(error, info) {
+    console.log('----------------------- error ----------------------')
+    console.log(error, info);
+  }
+
+  static getDerivedStateFromError(error) {
+    // Update state so the next render will show the fallback UI.
+    return { hasError: true }
+  }
+
+  handleRadio2Change = (e, field, value) => {
+    this.setState(prev => {
+      prev.formData[field] = value;
+      return prev;
+    }, () => {
+      this.props.updateForm(this.state.formData, field);
+      this.props.handleSecurityValidity(this.validate());
+      this.props.removeErrorMessage();
+    });
+  };
+
+  handleInputChange = (e) => {
+    const field = e.target.name;
+    const value = e.target.value;
+    this.setState(prev => {
+      prev.formData[field] = value;
+      return prev;
+    }, () => {
+      this.props.handleSecurityValidity(this.validate());
+      this.props.updateForm(this.state.formData, field);
+      this.props.removeErrorMessage();
+    })
+  };
+
+  validate = (field) => {
+    let pii = false;
+    let compliance = false;
+    let sensitive = false;
+    let accessible = false;
+    let isValid = true;
+    let textCompliance = false;
+    let textSensitive = false;
+    let textAccessible = false;
+
+    if (isEmpty(this.state.formData.pii)) {
+      pii = true;
+      isValid = false;
+    }
+    if (isEmpty(this.state.formData.compliance)) {
+      compliance = true;
+      isValid = false;
+    }
+
+    if (!isEmpty(this.state.formData.compliance)
+        && this.state.formData.compliance === "true"
+        && isEmpty(this.state.formData.textCompliance)) {
+      textCompliance = true;
+      isValid = false;
+    }
+    if (isEmpty(this.state.formData.sensitive)) {
+      sensitive = true;
+      isValid = false;
+    }
+    if (!isEmpty(this.state.formData.sensitive) && this.state.formData.sensitive === "true" && isEmpty(this.state.formData.textSensitive)) {
+      textSensitive = true;
+      isValid = false;
+    }
+    if (isEmpty(this.state.formData.accessible)) {
+      accessible = true;
+      isValid = false;
+    }
+    if (!isEmpty(this.state.formData.accessible) && this.state.formData.accessible === "true" && isEmpty(this.state.formData.textAccessible)) {
+      textAccessible = true;
+      isValid = false;
+    }
+    if (field === undefined || field === null || field === 3) {
```

Comment:
Can you explain where `field === 3` comes from? I think a comment here that explains why you need this check would be helpful.

Reply:
Removed `field === 3` since it's unused